src/fiatExchanges/quotes/FiatConnectQuote.ts

```diff
@@ -152,12 +160,70 @@ export default class FiatConnectQuote extends NormalizedQuote {
     return this.quoteResponseKycSchema?.kycSchema
   }
 
-  // TODO: Dynamically generate time estimation strings
-  getTimeEstimation(): SettlementTime {
-    // payment method can only be bank or fc mobile money
-    return this.getPaymentMethod() === PaymentMethod.Bank
-      ? SettlementTime.ONE_TO_THREE_DAYS
-      : SettlementTime.LESS_THAN_24_HOURS
+  getTimeEstimation(): SettlementEstimation {
+    const fiatAccountInfo = this.quote.fiatAccount[this.getFiatAccountType()]
+    if (!fiatAccountInfo?.settlementTimeUpperBound) {
+      // payment method can only be bank or fc mobile money
+      return this.getPaymentMethod() === PaymentMethod.Bank
+        ? DEFAULT_BANK_SETTLEMENT_ESTIMATION
+        : DEFAULT_MOBILE_MONEY_SETTLEMENT_ESTIMATION
+    }
+
+    const daysToSettlementEstimation = (
+      lowerBound: number,
+      upperBound: number
+    ): SettlementEstimation => {
+      if (lowerBound === upperBound || lowerBound === 0) {
+        return {
+          settlementTime: SettlementTime.LESS_THAN_X_DAYS,
+          upperBound,
+        }
+      }
+
+      return {
+        settlementTime: SettlementTime.X_TO_Y_DAYS,
+        lowerBound,
+        upperBound,
+      }
+    }
+
+    const hoursToSettlementEstimation = (
+      lowerBound: number,
+      upperBound: number
+    ): SettlementEstimation => {
+      if (upperBound <= 1) {
+        return {
+          settlementTime: SettlementTime.LESS_THAN_ONE_HOUR,
+        }
+      }
+
+      if (lowerBound === upperBound || lowerBound === 0) {
+        return {
+          settlementTime: SettlementTime.LESS_THAN_X_HOURS,
+          upperBound,
+        }
+      }
+
+      return {
+        settlementTime: SettlementTime.X_TO_Y_HOURS,
+        lowerBound,
+        upperBound,
+      }
+    }
+
+    const lowerBound = fiatAccountInfo.settlementTimeLowerBound ?? '0'
+    const lowerBoundInHours = Math.ceil(parseInt(lowerBound) / SECONDS_IN_HOUR)
```

Comment:
```suggestion
    const lowerBoundInHours = fiatAccountInfo.settlementTimeLowerBound ? Math.ceil(parseInt(fiatAccountInfo.settlementTimeLowerBound) / SECONDS_IN_HOUR) : 0
```
code golf. you don't need the extra operations if it's falsey to begin with